win32ss/printing/base/winspool/printers.c

```diff
@@ -384,10 +384,100 @@ EndPagePrinter(HANDLE hPrinter)
 }
 
 BOOL WINAPI
-EnumPrintersA(DWORD Flags, PSTR Name, DWORD Level, PBYTE pPrinterEnum, DWORD cbBuf, PDWORD pcbNeeded, PDWORD pcReturned)
+EnumPrintersA(DWORD Flags, PSTR pName, DWORD Level, PBYTE pPrinterEnum, DWORD cbBuf, PDWORD pcbNeeded, PDWORD pcReturned)
 {
-    TRACE("EnumPrintersA(%lu, %s, %lu, %p, %lu, %p, %p)\n", Flags, Name, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
-    return FALSE;
+    BOOL bReturnValue = FALSE;
+    DWORD cch;
+    PWSTR pwszName = NULL;
+    PSTR pszPrinterName = NULL;
+    PSTR pszServerName = NULL;
+    DWORD i;
+    PPRINTER_INFO_4W ppi4w = NULL;
+    PPRINTER_INFO_4A ppi4a = NULL;
+
+    TRACE("EnumPrintersA(%lu, %s, %lu, %p, %lu, %p, %p)\n", Flags, pName, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
+
+    if (pName)
+    {
+        // Convert pName to a Unicode string pwszName.
+        cch = strlen(pName);
```

Comment:
Using MultiByteToWideChar/WideCharToMultiByte to determine the number of characters really is better (especially in the W->A case), but for consistency's sake, this way is fine here.

Reply:
"... this way is fine here." OK.